src/wp-includes/media.php

```diff
@@ -2081,25 +2081,29 @@ function wp_sizes_attribute_includes_valid_auto( string $sizes_attr ): bool {
 }
 
 /**
- * Prints a CSS rule to fix potential visual issues with images using `sizes=auto`.
+ * Enqueues a CSS rule to fix potential visual issues with images using `sizes=auto`.
  *
  * This rule overrides the similar rule in the default user agent stylesheet, to avoid images that use e.g.
  * `width: auto` or `width: fit-content` to appear smaller.
  *
- * @since 6.7.1
+ * @since 6.9.0
+ *
  * @see https://html.spec.whatwg.org/multipage/rendering.html#img-contain-size
  * @see https://core.trac.wordpress.org/ticket/62413
  */
-function wp_print_auto_sizes_contain_css_fix() {
+function wp_enqueue_img_auto_sizes_contain_css_fix(): void {
```

Comment:
Shouldn't we, either within this function, or in some other logic that runs on a hook shortly before this, check for `has_action( 'wp_head', 'wp_print_auto_sizes_contain_css_fix' )`?

I think we should include this for BC, to make sure sites that removed the original hook don't see adverse effects now because we changed the hook. This is similar to how we did it elsewhere before I think.

Reply:
@felixarntz Thanks for raising this again. After re-examining the purpose of the `wp_img_tag_add_auto_sizes` filter, I realize I was wrong to suggest that it was the right way to prevent the style from being printed. See https://github.com/WordPress/wordpress-develop/pull/8954#issuecomment-3364076685